From 826fbc96d56fa4ea4e89d6870cc97cfc3aa99f34 Mon Sep 17 00:00:00 2001
From: Aaron Rosen <aaronorosen@gmail.com>
Date: Wed, 16 Apr 2014 11:31:25 -0700
Subject: [PATCH] netaddr<=0.7.10 raises ValueError instead of AddrFormatError

This patch ensures that ValueError is also caught in addition to
AddrFormatError as in netaddr>=0.7.11 AddrFormatError is raised and
in netaddr<=0.7.10 ValueError is raised.

Change-Id: I595c90e42129a2d365f3860e3042e826bd031365
Closes-bug: #1308675
(cherry picked from commit dd2ccd12adfb6c7d4aa500ba4120f767beeed35c)
(cherry picked from commit 42a8539d497322716df0150c2123befd246d69d8)
---
 neutron/extensions/securitygroup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/neutron/extensions/securitygroup.py b/neutron/extensions/securitygroup.py
index 637dbe3..37031e2 100644
--- a/neutron/extensions/securitygroup.py
+++ b/neutron/extensions/securitygroup.py
@@ -164,7 +164,7 @@ def convert_ip_prefix_to_cidr(ip_prefix):
     try:
         cidr = netaddr.IPNetwork(ip_prefix)
         return str(cidr)
-    except (TypeError, netaddr.AddrFormatError):
+    except (ValueError, TypeError, netaddr.AddrFormatError):
         raise qexception.InvalidCIDR(input=ip_prefix)
 
 
